core/controllers/question_editor_test.py

```diff
@@ -364,10 +305,28 @@ def test_put_with_admin_email_allows_updation(self):
         self.login(self.ADMIN_EMAIL)
         csrf_token = self.get_new_csrf_token()
         _ = self.put_json(
-            '%s/%s/%s' % (
-                feconf.QUESTION_SKILL_LINK_URL_PREFIX, self.question_id,
-                self.skill_id
-            ), {'new_difficulty': 0.9}, csrf_token=csrf_token)
+            '%s/%s' % (
+                feconf.QUESTION_SKILL_LINK_URL_PREFIX, self.question_id
+            ), {
+                'new_difficulty': 0.9,
+                'action': 'update_difficulty',
+                'skill_id': self.skill_id
+            }, csrf_token=csrf_token)
+
+        _ = self.put_json(
+            '%s/%s' % (
+                feconf.QUESTION_SKILL_LINK_URL_PREFIX, self.question_id
+            ), {
+                'difficulty': 0.9,
+                'action': 'edit_links',
+                'skill_ids_task_list': [{
+                    'id': 'skill_2',
+                    'task': 'add'
+                }, {
+                    'id': 'skill_2',
+                    'task': 'remove'
```

Comment:
The IDs of the skills should be checked after this.

Also add a test where you add a different skill and remove an existing one.

Also add a test where you try to remove the only skill that exists (it should error).

Reply:
The last one is just a frontend check for that modal, since if we want to actually delete a question from the frontend, then this should happen, since the Unlink Question From Skill button in the questions table also uses this same endpoint.

Reply:
FIrst 2, done.